### ISSUE TITLE
Mark Some Job Parameters as Sensitive, Requiring Re-entry to Re-run

### ISSUE THREAD

**Issue by bryanculver** [nautobot/nautobot#2096]
### As ...

Tammy - Tools Integrator

### I want ...

To mark input parameters as sensitive without having to mark the entire job as sensitive, minimizing exposure of just those parameters to all users who can view JobResults. After #2091, the parameters of these jobs will be entirely excluded from the database if the job is marked "sensitive", completely eliminating the ability to rerun, where a bulk of user input can be in non-sensitive fields.

### So that ...

I can write and use jobs that access other systems, etc. using per-user authentication credentials for those other systems as a job input.

### I know this is done when...

- A pattern exists for marking specific Job parameters as sensitive (perhaps adding a `PasswordVar` job variable type?). Such parameters must NOT be saved in job_kwargs.

### Optional - Feature groups this request pertains to.

- [X] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [ ] Plugins (and other Extensibility)
- [X] Security (Secrets, etc)
- [ ] Image Management
- [ ] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

**Comment by itdependsnetworks** [comment]
@bryanculver is this complete?

**Comment by bryanculver** [comment]
Not entirely. We can mark a whole job as "sensitive" as opposed to discrete variables. This makes re-running a Job that is marked sensitive a bit more complicated as you must re-enter all values.

A contrived example would be a Job to push a config to a device, I may require to enter a change ID. I would want to prevent a user from just reusing the change ID entered from a previous run, so I could mark that specific one as sensitive (it's not a Secret so it makes sense to not make it so).

Ultimately, I could see this as a "nice to have" enhancement, and say "as long as you use Secrets you can mark a job as 'not sensitive'".

An alternative approach above would be to use something like `PasswordVar` or `SensitiveVar` which would prevent logging or storage in the DB. Or quite possibly use some sort of hashing to balance flexibility between "Must be a Secret" or "Must be sensitive Job"
